rekor_entries.proto

```diff
@@ -0,0 +1,163 @@
+     */
+    rpc CreateLogEntry(CreateLogEntryRequest) returns (LogEntry){
+        option (google.api.http) = {
+          post: "/api/v2/log/entries"
+          body: "*"
+        };
+        option (google.api.method_signature) = "log_entry";
+    }
+    /**
+     * Retrieves an entry and inclusion proof from the transparency log (if it exists)
+     */
+    rpc GetLogEntry(GetLogEntryRequest) returns (LogEntry){
+        option (google.api.http) = {
+          get: "/api/v2/log/entries/{uuid}"
+          additional_bindings: {
+              get: "/api/v2/log/entries"
+          }
+        };
+    }
+    /**
+     * Returns a list of (optionally filtered) entries from the transparency log
+     */
+    rpc ListLogEntries(ListLogEntriesRequest) returns (ListLogEntriesResponse){
+        option (google.api.http) = {
+          get: "/api/v2/log/entries/retrieve"
+        };
+    }
+}
+
+message CreateLogEntryRequest {
+    /*
+     * The proposed entry to be made into the log
+     */
+    dev.sigstore.rekor.v2.types.ProposedLogEntry proposed_entry = 1 [(google.api.field_behavior) = REQUIRED];
+}
+
+message LogEntry {
+    /*
+     * The log ID, which is computed as the SHA256 digest of the DER-encoded public key of the log
+     */
+    string log_id                               = 1;
+    /*
+     * The index of the entry within the log
+     */
+    int64 log_index                             = 2;
+    /*
+     * The fields of the entry
+     */
+    dev.sigstore.rekor.v2.types.TypedEntry body = 3;
+    /*
+     * The time that the entry was integrated into the transparency log
+     */
+    int64 integration_time                      = 4;
+    /*
+     * The attestation provided at the time of insertion (if present)
+     */
+    Attestation attestation                     = 5;
+    /*
+     * The proof that the entry exists in the transparency log
+     */
+    Verification verification                   = 6;
+}
+
+message Attestation {
+    /*
+     * The attestation content
+     */
+    bytes payload = 1;
+}
+
+message Verification {
+    /*
+     * The proof that the entry exists in the transparency log
+     */
+    InclusionProof inclusion_proof = 1;
+    /*
+     * The proof that the entry exists in the transparency log
+     */
+    bytes signed_entry_timestamp   = 2;
+}
+
+message InclusionProof {
+    /*
+     * The index of the entry within the log
+     */
+    int64 log_index        = 1;
+    /*
+     * The hash value stored at the root of the merkle tree at the time the proof was generated
+     */
+    string root_hash       = 2;
+    /*
+     * The current number of nodes in the merkle tree
+     */
+    int64 tree_size        = 3;
+    /*
+     * One or more hash values that can be combined to calculate an inclusion proof
+     */
+    repeated string hashes = 4;
+}
+
+message GetLogEntryRequest {
+    /*
+     * The universally unique identifier for the entry within the log shard
+     */
+    string uuid     = 1;
```

Comment:
nit: consider renaming uuid to entryID to indicate that we want the universal identifier

Reply:
if we're assuming that the v2 API will only be run with rekor server >=0.6, then we can probably rename all mentions of uuid to entryID?

Reply:
yeah, I thought about that as well as normalizing "tree" vs "shard" and trying to make it consistent.